src/server.rs

```diff
@@ -509,7 +512,21 @@ impl RollupBoostServer {
 
         let (l2_payload, builder_payload) = tokio::join!(l2_client_future, builder_client_future);
         let payload = match (builder_payload, l2_payload) {
-            (Ok(builder), _) => Ok(builder),
+            (Ok(builder), Ok(l2)) => {
+                let builder_gas_used = builder
+                    .0
+                    .execution_payload
+                    .payload_inner
+                    .payload_inner
+                    .gas_used;
+                let l2_gas_used = l2.0.execution_payload.payload_inner.payload_inner.gas_used;
+                if builder_gas_used >= l2_gas_used * self.l2_boost_factor {
```

Comment:
you're multiplying by an integer here, not a float, so I think we have incorrect behaviour.